configs/common/packages.yaml

```diff
@@ -20,7 +20,7 @@
       version: [1.78.0]
       variants: ~atomic +chrono +date_time +exception +filesystem ~graph ~iostreams ~locale ~log ~math ~mpi ~numpy +pic +program_options +python ~random +regex +serialization ~signals +system +test +thread +timer ~wave cxxstd=14 visibility=hidden
     bufr:
-      version: [11.7.1]
+      version: [12.0.0]
```

Comment:
I think I'm good with making burf@12.0.0 the default version, but I want to double check. JEDI users should be using spack-stack-1.4.0 which has bufr@11.7.1 installed. Because of this JEDI users will be insulated from this change so it's okay to make bufr@12.0.0 the default version on the develop branch.

Reply:
Correct

Reply:
just a note: bufr@12.0.0 is producing large differences in GSI regression tests (as reported by @DavidHuber-NOAA: https://github.com/NOAA-EMC/GSI/issues/589#issuecomment-1635811968). fine to merge this here, but i expect there might be some issues w/ some applications during transition to this version.

Reply:
@ulmononian one thing to check would be if the GSI is currently using the bufr@11.x bufr_8 component (which is using 8-byte integers and 8-byte floats, ie double precision), or the bufr_d component (which has 4-byte integers and 8-byte floats). See the README.md notes at https://github.com/NOAA-EMC/NCEPLIBS-bufr/tree/bufr_v11.7.1 for details.

The only component available in bufr@12.0.0 is bufr_4 which is 4-byte integers and 4-byte floats (ie single precision) so the GSI bufr related code could be making a switch from double to single precision.

Reply:
Let me note that if it turns out that we need both bufr@12 and bufr@11 in spack-stack-1.5.0, we can install both and configure the different environments (gsi-env etc.) such that they load the correct one.

Reply:
@srherbener Indeed, the GSI uses the bufr_d library, so that could explain the differences.  Thanks!

Reply:
@srherbener thanks for providing that insight, steve. @DavidHuber-NOAA do you expect the GSI team to accept the differences w/ bufr_4 / bufr@12 or should we plan to provide bufr@11 for gsi purposes for the foreseeable future?

Reply:
@ulmononian I don't expect that they will accept it or at least that they will want to run some more testing before making the switch.  Also, I don't see any hints on their GitHub page that they desire to upgrade at this time.  But I have asked the question -- and will ask again on a different channel -- and will update when I hear back.